accepted/00000-replace-salt-toaster.md

```diff
@@ -0,0 +1,138 @@
+our team that we want to take advantage of.
+
+# Detailed design
+[design]: #detailed-design
+
+<!-- This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with the product to understand, and for somebody familiar with the internals to implement. -->
+
+<!-- This section should cover architecture aspects and the rationale behind disruptive technical decisions (when applicable), as well as corner-cases and warnings. Whenever the new feature creates new user interactions, this section should include examples of how the feature will be used. -->
+
+## Overview
+
+The Salt upstream testsuite is executed in virtual machines (VMs) that run the
+different operating systems we support. These VMs are set up with sumaform and
+salt-jenkins. Terracumber triggers terraform to create the VMs from within a
+Jenkins job and launches the testsuite, connecting to each VM over SSH.
+
+## Provisioning Test VMs with Sumaform + Salt-Jenkins
+
+A new module "salt\_testenv" is added to sumaform. This module defines a new
+role, "salt\_testenv", and the default values needed to run all testsuites, e.g.
+4GiB of RAM. Sumaform uses this role to apply the `golden-images-provision.sls`
+state file from the Salt-Jenkins repository.
+
+Up to now, all state files used by sumaform are maintained in the sumaform git
+repository. For provisioning the salt\_testenv VMs, we don't want to maintain
+our own states that define which dependencies need to be installed on what
+operating system. Salt upstream already maintains such states for their own test
+infrastructure. We re-use their states by adding `salt-jenkins` as a git
+submodule to sumaform and merge the salt state trees.
+
+In order to avoid future state id clashes, all existing state ids will get a
+"sumaform-" prefix.
+
+TODO: expand the merge mechanism
+
+## Terracumber
+
+Terracumber is a tool used by our QE squad to operate terraform and cucumber for
+their regular test suite execution. The same tool can be extended to set up the
+required VMs and start the Salt test suite execution.
+
+## Executing the Salt Test Suite
+
+The test suite uses `nox` to set up a Python virtual environment with all
+required dependencies. Arbitrary CLI flags for `pytest`, the used Python testing
+framework, can be passed through. We use this for two purposes: passing the path
+to exactly those test groups that we want to run, and to skip unstable tests.
+
+### Test Script
+
+A test script that passes the correct arguments makes it easy to run the same
+test selection in different environments: locally, in Jenkins or in GitHub
+Actions on Pull Requests. The script is simple on purpose. It takes a
+configuration file that maps test group names to the test locations and includes
+a skip list, one for each test group.
+
+Example:
+```sh
+$ cat test_config.yml
+groups:
+  unit:
+    tests_dirs:
+      - tests/pytests/unit
+      - tests/unit
```

Comment:
Does it mean we are going to use it for `unit` tests only and avoid running `functional` and `integration` tests?

Reply:
No, we will run all four test groups (unit, functional, integration, scenario). This is just a short example of an incomplete config to show the desired input and output of the (to-be-developed) test execution script.

Reply:
Thanks for explanation :+1: